src/components/tree/phyloTree/types.ts

```diff
@@ -0,0 +1,260 @@
+import { NODE_NOT_VISIBLE, NODE_VISIBLE_TO_MAP_ONLY, NODE_VISIBLE } from "../../../util/globals";
+import { PhyloTree } from "./phyloTree";
+
+export type StringLayouts = "scatter"
+
+export type NumericLayouts = "rect" | "radial" | "unrooted" | "clock"
+
+export type Layout = NumericLayouts | StringLayouts
+
+export type Distance = "num_date" | "div"
+
+export type TreeElement = 
+  ".branch.S" |
+  ".branch.T" |
+  ".branch" |
+  ".branchLabel" |
+  ".conf" |
+  ".grid" |
+  ".regression" |
+  ".tip" |
+  ".tipLabel" |
+  ".vaccineCross" |
+  ".vaccineDottedLine"
+
+export type Visibility = typeof NODE_NOT_VISIBLE | typeof NODE_VISIBLE_TO_MAP_ONLY | typeof NODE_VISIBLE
+
+export interface Colorings {
+  [key: string]: {
+    type: string
+  }
+}
+
+export interface Genotype {
+  gene: string
+  positions: number[]
+  aa: boolean
+}
+
+export type ScaleType = "ordinal" | "categorical" | "continuous" | "temporal"
+
+export interface ColorScale {
+  colorBy: string
+  continuous: boolean
+  domain?: any[]
+  genotype: Genotype | null
+  legendBounds?: Record<number | "unknown", [number, number]>
+
+  /** A map of legendValues to a value for display in the legend. */
+  legendLabels?: Map<any, any>
+
+  /** An array of values to display in the legend. */
+  legendValues: any[]
+
+  scale: (value: any) => string
+  scaleType: ScaleType | null
+  version: number
+  visibleLegendValues: string[]
+}
+
+export interface TemporalConfidence {
+  exists: boolean
+  display: boolean
+  on: boolean
+}
+
+export interface ReactTreeNode {
+  existingFilterState?: "active" | "inactive" | null
+  idx?: number
+  isBranch: boolean
+  name?: string
+  node?: PhyloNode
+  treeId?: string
+}
+
+export interface ReduxNode {
+  /** the index of the node in the nodes array. set so that we can access visibility / nodeColors if needed */
+  arrayIdx?: number
+
+  branch_attrs?: {
+    mutations?: {
+      [gene: string]: string[]
+    }
+    labels?: Record<string, any>
+  }
+  children?: ReduxNode[]
+  currentGt?: number
```

Comment:
[to check] `currentGt` is (or at least can be) an array (e.g. `node.currentGt = [...newState]`). But perhaps this is the same property name on a different object?

Reply:
I see: https://github.com/nextstrain/auspice/blob/2af5974e4631a6768727b2a21bae1182bb354047/src/util/setGenotype.js#L28

That file hasn't been typed yet. Changing the type to `currentGt?: number | number[]` doesn't work as-is because it is assumed to be a scalar for `PhyloNode.px`:

https://github.com/nextstrain/auspice/blob/2af5974e4631a6768727b2a21bae1182bb354047/src/components/tree/phyloTree/layouts.ts#L95

Do you think it's worth addressing in this PR? I could also just leave a comment that the type is incomplete.

Reply:
Two points:

> That file hasn't been typed yet. Changing the type to currentGt?: number | number[] doesn't work as-is because it is assumed to be a scalar for PhyloNode.px:

Within that function we create an array `node.currentGt = [...newState];` then convert to a string `nodes.forEach((n) => {n.currentGt = n.currentGt.join(' / ');});`. I _think_ everywhere else in the code expects a string. So I guess keeping the type as `currentGt?: str` will flag this up when we eventually type this file? (And at that point the code should be fixed!)

Secondly, the `currentGt` is (AFAICT) always a string, not a number.